File: Code/Laravel-Backend/app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Product extends Model
{
    protected $primaryKey = 'code';
    protected $keyType = 'string';
    public $incrementing = false;

    protected $fillable = ['code', 'name', 'price', 'tax_rate'];

    protected $casts = [
        'price' => 'decimal:2',
        'tax_rate' => 'decimal:4',
    ];
}

File: Code/DotNet-Backend/Models/Product.cs
namespace BillingApi.Models;

public class Product
{
    public string Code { get; set; } = "";
    public string Name { get; set; } = "";
    public decimal Price { get; set; }

    /// <summary>e.g. 0.18 for 18% GST</summary>
    public decimal TaxRate { get; set; }
}
